Polished component and especially slot content (#397)
withastro/docs: src/pages/en/core-concepts/astro-components.md

### Slots

> Astro components can accept HTML content in your component template. This "child" content will be injected into its `<slot />` element.

Reviewer: Agree that this needs some work — I find that first sentence hard to parse. “HTML content in your component template?” Sure my component template is full of HTML content! 🤔 

How about something like

```suggestion
If you want to pass HTML content into your component template, you can use the `<slot />` element to control where that “child” content is rendered.
```

> This pattern is the basis of an Astro layout component: an entire page of HTML content can be “wrapped” with `<Layout></Layout>` tags and sent to the Layout component to render inside of common page elements.

Reviewer 2: ```suggestion
```
Move this to a better location under the code block

Author: Yes, I had been thinking about this.

```astro
---
// src/pages/index.astro
import Wrapper from '../components/Wrapper.astro';
---
<Wrapper name="Astro">
  <h2>I am a person.</h2>
  <p>Here is some stuff about me.</p>
</Wrapper>

// src/components/Wrapper.astro

> ---

Reviewer: I guess there’s also a question about whether it would more helpful to see `Wrapper` first and then the page that uses it?

> ---

Author: Both things I'd thought about.
a. They were together initially, so the formatting issue has always been there, but I have no issues splitting them up.

b. This was an explicit choice to swap the order (not saying it's right, just saying it was an intentional choice) and I'll add a comment below to explain the thinking behind this change before going point by point through all the (great, helpful!) comments.  :)

> ---

Reviewer: Got it! I’m not that concerned on the order question if it’s been thought about, but do think the formatting change would be good. Team work! 🙌

  <slot />  <!-- children will go here -->
  <Footer />
</div>
```

> 

Reviewer 2: ```suggestion
This pattern forms the basis of an Astro layout component. Whereby different islands can be placed within your `<Layout>` along side HTML markup that are coming from external sources. To do this we place `<slots>` and *named* slots `<slots name="">` within our layout to direct where the content goes. 
```

#### Named Slots

> You can pass multiple pieces of content to your Astro component, and place them precisely, using named slots. 

Reviewer 2: ```suggestion
You can precisely pass multiple pieces of content directly into your Astro component, by *naming* the slots. 
```

Reviewer: I prefer the existing wording here — “passing precisely” seems confusing to me. But maybe the emphasis is helpful: `using **named slots**`?

Reviewer 2: Still got to finish reviewing this, 😬
